tephra-hbase-compat-1.1-base/src/main/java/org/apache/tephra/hbase/txprune/CompactionState.java

```diff
@@ -75,18 +76,34 @@ public void record(CompactionRequest request, @Nullable TransactionVisibilitySta
   }
 
   /**
+   * Stops the current {@link PruneUpperBoundWriter}.
+   */
+  public void stop() {
+    if (pruneUpperBoundWriter != null) {
+      pruneUpperBoundWriter.stop();
+    }
+  }
+
+  /**
    * Persists the transaction state recorded by {@link #record(CompactionRequest, TransactionVisibilityState)}.
    * This method is called after the compaction has successfully completed.
    */
   public void persist() {
     if (pruneUpperBound != -1) {
-      try {
-        dataJanitorState.savePruneUpperBoundForRegion(regionName, pruneUpperBound);
-        LOG.debug(String.format("Saved prune upper bound %s for region %s", pruneUpperBound, regionNameAsString));
-      } catch (IOException e) {
-        LOG.warn(String.format("Cannot record prune upper bound in table %s after compacting region %s",
-                               stateTable, regionNameAsString), e);
-      }
+      PruneUpperBoundWriter upperBoundWriter = getPruneUpperBoundWriter();
```

Comment:
There are redundant instances of class `PruneUpperBoundWriter` => `upperBoundWriter` and a`pruneUpperBoundWriter`